### ISSUE TITLE
dice: .choose output is messy

### ISSUE THREAD

**Issue by dgw** [sopel-irc/sopel#1420]
I started this thought process after noticing that `.choose thing1, thing2, thing3` will yield choices that include the leading space (`thing1`, ` thing2`, and ` thing3`), but it's not that simple.

Since `.choose` just uses a simple `str.split()` as-is, without cleaning the values at all, it's already starting with messy data. Then it swaps the delimiter used for no particular reason ([comment claims](https://github.com/sopel-irc/sopel/blob/d73a410717d6cb2b164bf5411f20a995210bbc22/sopel/modules/dice.py#L248) it's "to prevent ambiguity", but I think it actually does the opposite).

What comes out the other end:

```
<dgw> .choose thing1, thing2, thing3
<Sopel> dgw: Your options: thing1|  thing2|  thing3. My choice:  thing2
```

Extra spaces everywhere, delimiters that don't really make sense with the spacing around them… Ugly.

Current plans to sort this out:

  - [x] **In 6.6.0:** Clean the input to avoid extra spaces in the output. ~~Maybe switch to always use `, `&nbsp;as delimiter in output.~~ [#1421]
  - [x] **In 7.0.0:** ~~Drop support for everything but `,` (and maybe `/`) in the input, and~~ always use `, `&nbsp;in output ~~(if not already changed for 6.6.0)~~. [#1425]

There's considerable lead time before Sopel 7, so any user objections to dropping `\` and `|` as supported delimiters may be dropped in the comments here, or in the comments of the PR I will open with the change some time in early 2019, after 6.6.0 is out.

**Comment by alanhuang122** [comment]
I noticed the extra spaces issue ~9M ago and fixed it with a one-liner :v https://github.com/alanhuang122/sopel/commit/97d8ae65c743638c7e8404295599e188004dd434.

I think using commas as exclusive delimiters in output (and quoting choices with commas internally) would work.

In my experience, the current system of splitting on `|\/,` works fine - many of my users
`.choose course of action 1, 2, 3 | course of action foo, bar, baz` - response is
`$nick: Your options: course of action 1, 2, 3/ course of action foo, bar, baz. My choice: course of action 1, 2, 3`

Yet another reason to just use commas - even non-comma delimiters get comma rules applied.

**Comment by dgw** [comment]
That one-liner is exactly what I'm planning to do, haha. Think I'll just cherry-pick that commit, if you don't mind. :cat: Or you can make a PR yourself by doing the same!

Swapping delimiters around is what really gets me about the current code. It's just…ugly, most of the time, primarily due to spacing. The fact that you can put in `.ch option1, option2, option3` and get out `Your options: option1| option2| option3` (even with the `strip()` patch) is kind of absurd. Pipes and slashes look strange if the surrounding whitespace is imbalanced.

Maybe I won't take out the extra input separators… Maybe just making the output look better is enough. Maybe that can even be done for 6.6.0 (except I keep adding more and more to it, and it's already probably not coming out until next year because of that :joy_cat:).

**Comment by dgw** [comment]
Ooh, I didn't even notice the filename. Whoops. No excuse for that; I'm not even tired… 🤭 Fine then, I'll just do the same thing (as originally planned) in my own PR.

**Comment by Exirel** [comment]
I was looking at the output part: https://github.com/dgw/sopel/blob/b869fb590c760aefb579de52629fed94bbd00d32/sopel/modules/dice.py#L256-L263

Maybe something like this could improve how it's displayed?

```
output_delimiters = [
    (',', '{delimiter} '),
    ('|', ' {delimiter} '),
    ('/', ' {delimiter} '),
    ('\\', ' {delimiter} '),
]
show_delim = ' '
for delimiter, template in output_delimiters:
    if delimiter not in trigger.group(2):
        show_delim = template.format(delimiter=delimiter)
        break
```

This way it would output:

```
<dgw> .choose thing1, thing2,thing3
<Sopel> dgw: Your options: thing1 | thing2 | thing3. My choice: thing2
<dgw> .choose thing1| thing2 |thing3
<Sopel> dgw: Your options: thing1, thing2, thing3. My choice: thing2
```

I could probably make a PR or something if you wish.

**Comment by dgw** [comment]
Whitespace stripping (#1421) has been merged. Further improvements will be made in Sopel 7, so moving this issue to the new milestone.

**Comment by dgw** [comment]
@Exirel What would you say to implementing what @alanhuang122 suggested [above](https://github.com/sopel-irc/sopel/issues/1420#issuecomment-445104845) (output `,` between options no matter what the input used, and quote values that contain it)?

Could be part of your existing PR or a separate one, whatever works.

**Comment by Exirel** [comment]
By the way, my PR #1425 is available here for review after 6.6.0 is released.
